aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -864,10 +864,10 @@ void add_out_dense_sparse_csr_cpu(
     return;
   }
 
-  auto valuesBuffer = src_values.to(commonDtype).view({-1, src_values.size(-1)});
+  auto valuesBuffer = src_values.to(commonDtype).reshape({-1, src_values.size(-1)});
   resultBuffer = resultBuffer.view({-1, out.size(-2), out.size(-1)});
-  auto src_crow_indices = src.crow_indices().view({-1, src.crow_indices().size(-1)});
-  auto src_col_indices = src.col_indices().view({-1, src.col_indices().size(-1)});
+  auto src_crow_indices = src.crow_indices().reshape({-1, src.crow_indices().size(-1)});
+  auto src_col_indices = src.col_indices().reshape({-1, src.col_indices().size(-1)});
```

Comment:
The CPP version uses just `reshape` while the CUDA version is doing `reshape + contiguous`. Either way, if `reshape` is called (or `flatten`), the batch dimensions become regular, and there is no need in sophisticated maps from a linear index to a multi-dim offset.
It seems like dropping `reshape` is best for performance, and we could use `make_offset_calculator` to keep the CPU and GPU kernel relatively intact. Otherwise it is best to re-implement these kernels and use `TensorIterator.for_each`.

Reply:
> The CPP version uses just reshape while the CUDA version is doing reshape + contiguous

This is due to CPU version using accessor indexing while the CUDA version uses pointer indexing.

The rest of the comments are OT to this PR but I agree that the CPU and GPU kernels could be unified here.

Reply:
Why are they different? They use the same code...

Reply:
Sorry, @pearu, I thought these are inside invariant checks, but they turn out to be in the `add` op. Apologies for the noise...